apps/data/src/lib/enrich/enrich/macro_execution.py

```
@@ -0,0 +1,136 @@
+
+        Input DataFrame columns: id, macro_id, data
+        Output DataFrame columns: result (JSON string | None), error (string | None)
+        """
+        results = [None] * len(pdf)
+        errors = [None] * len(pdf)
+
+        # Build items list for the backend
+        items = []
+        idx_map = []  # maps backend item index -> original DataFrame index
+
+        for pos, (df_idx, row) in enumerate(pdf.iterrows()):
```

Ruff (0.15.10)

[warning] Loop control variable `df_idx` not used within loop body

(B007)